src/views/AddItem.jsx

```diff
@@ -12,24 +12,56 @@ const dayConverter = (text) => {
 	}
 };
 
-export function AddItem({ listId }) {
+export function AddItem({ listId, data }) {
 	const [itemName, setItemName] = useState('');
 	const [frequency, setFrequency] = useState('soon');
-	const [message, setMessage] = useState('');
+	const [itemMessage, setItemMessage] = useState('');
+	const [errorMessage, setErrorMessage] = useState('');
+
+	const clearMessage = (messageToAdjust) => {
+		setTimeout(() => {
+			messageToAdjust('');
+		}, 3000);
+	};
 
 	const handleSubmit = async (e) => {
 		e.preventDefault();
+
+		if (!itemName.trim()) {
+			setErrorMessage('Please enter item name.');
+			clearMessage(setErrorMessage);
+			return;
+		}
+
+		const existingItem = data.find((item) => {
+			return (
+				itemName.localeCompare(item.name, 'en', {
+					sensitivity: 'base',
+					ignorePunctuation: true,
+				}) === 0
+			);
```

Comment:
I have never used `itemName.localeCompare` before!

@tialaaa @ismarjiw  Why did you take this path, I'm curious!